test/test_helpers.py

```diff
@@ -74,6 +76,101 @@ def fake_script_calls(test_case: unittest.TestCase,
     return calls  # type: ignore
 
 
+def create_framework(
```

Comment:
This is ported from the other branch, only used in `test_charm.py`. In `test_framework.py`, it uses a different function to create a framework because a model also needs to be passed in.